src/Runner.Common/JobServer.cs

```diff
@@ -56,6 +67,40 @@ public async Task ConnectAsync(VssConnection jobConnection)
             _hasConnection = true;
         }
 
+        private async Task CheckHealthEndpointsAsync()
+        {
+            try
+            {
+                Trace.Info("Requesting Actions Service health endpoint status");
+                // Construct the HttpClient object directly, without going through GetClient<T>
+                // as we haven't properly connected to the server at this point
+                var baseUri = new Uri(_connection.Uri.GetLeftPart(UriPartial.Authority));
+                var client = new HealthHttpClient(baseUri, _connection.InnerHandler, false);
```

Comment:
I would suggest just create `HttpClient` inline as you did for the github api endpoint.